torchao/prototype/low_bit_optim/adam.py

```diff
@@ -109,6 +109,7 @@ def step(self, closure=None):
 
 # this will work with any optim state tensor subclass that implements aten.lerp.Scalar and aten.copy_.default
 # and param tensor subclass that implements aten.add_.Tensor, and aten.addcdiv_.default
+# NOTE: should we cast inputs to FP32 to ensure computations are always in FP32?
```

Comment:
i override a few methods to cast the input to the weight dtype because the Flux model occasionally upcasts things to fp32 inside layernorm. you are saying fp32 is more correct than bf16?

Reply:
For optimizer step, internal calculations should be done in FP32 to ensure accurate results.